server.go

```diff
@@ -183,13 +204,16 @@ func parseURLParameters(url, pattern string) map[string]string {
 		return nil
 	}
 
-	parameters := make(map[string]string)
+	parameters := make(map[string]string, len(patternSegments))
 
 	for i, segment := range patternSegments {
 		if len(segment) > 2 && segment[0] == '{' && segment[len(segment)-1] == '}' {
 			paramName := segment[1 : len(segment)-1]
 			parameters[paramName] = urlSegments[i]
-		} else if segment != urlSegments[i] {
+			continue
+		}
+
+		if segment != urlSegments[i] {
```

Comment:
Some go developer say, that you should not use `else` in go. They try to return early.